tools/python/boutiques/descriptor2func.py

```diff
@@ -0,0 +1,27 @@
+from boutiques.util.utils import loadJson
+from boutiques import validate, execute, prettyprint
+import json
+
+
+def function(descriptor, name=None, mode='launch'):
+
+    validate(descriptor)
+    descriptor_json = loadJson(descriptor)
+
+    def f(*args, **kwargs):
+        if mode == 'launch':
+            return execute(mode, descriptor,
+                           json.dumps(kwargs), *args)
+        elif mode == 'simulate':
+            return execute(mode, descriptor,
+                           '-i', json.dumps(kwargs), *args)
+        else:
+            raise Exception("Unsupported mode: {}".format(mode))
+
+    if name is None:
+        f.__name__ = str(descriptor_json['name'])
+    else:
+        f.__name__ = name
+    f.__doc__ = prettyprint(descriptor)
```

Comment:
Since this is a lot of text, might be worth pruning down to the last bit that is the input params? We could just grep the heading for them and do after that. I don't have a strong feeling here, but that may make the helptext more easily parseable, esp for big descriptors.

Reply:
yeah, I wouldn't like to have to post-process pprint here. Maybe we could add various options to pprint to show the different sections.